### ISSUE TITLE
zip with password

### ISSUE THREAD

**Issue by alliesaizan** [r-lib/zip#38]
Hello,

I would like to compress a file with a password in R. Could this functionality be added to the zip or zipr function?

Thanks!

**Comment by gaborcsardi** [comment]
Possibly, yes. I'll need to look around a bit to see what exactly is supported by other zip programs, so that we are compatible with them.

**Comment by alliesaizan** [comment]
Thank you! Looking forward to hearing what you find.

**Comment by Matherion** [comment]
Yes, this would be awesome! This would be great for ethical data sharing (i.e. respecting participants' privacy and sharing as much data as possible). Implementation of 256-bit AES encryption would allow anonymization and preservation of raw data to be handled programmatically. That could really help people to share their data. (... And there are loads of other reasons why being able to encrypt stuff would be brilliant, of course :-))

**Comment by danielinteractive** [comment]
+1

**Comment by gaborcsardi** [comment]
Unfortunately miniz does not support passwords, we'd need to switch to https://github.com/zlib-ng/minizip-ng for that. Which is not so easy: https://github.com/r-lib/zip/issues/122#issuecomment-4601583813

I am going to leave this open for now. Please add a thumbs up to the first comment if you want this implemented in the future.

**Comment by gaborcsardi** [comment]
I think we can add our own crypto on top of miniz's low level functions.
